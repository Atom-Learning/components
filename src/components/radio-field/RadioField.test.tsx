import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { RadioField } from '.'

describe("RadioField component", () => {
  it('renders', async () => {
    const { container } = render(<RadioField />)

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(<RadioField />)

    expect(await axe(container)).toHaveNoViolations()
  })
})