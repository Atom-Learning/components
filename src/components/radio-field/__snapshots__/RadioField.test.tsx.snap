// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioField component renders 1`] = `
.sx03kze-2grog {
  margin-right: var(--sx-space-3);
  transform: translateY(var(--sx-space-0));
}

.sxh0wga {
  color: var(--sx-colors-tonal800);
  display: block;
  font-family: var(--sx-fonts-body);
  font-weight: 600;
  margin: 0;
}

.sxh0wgaryo8u--size-md {
  font-size: var(--sx-fontSizes-md);
  line-height: 1.5;
}

.sxh0wgaryo8u--size-md::before {
  content: '';
  margin-bottom: -0.3864em;
  display: table;
}

.sxh0wgaryo8u--size-md::after {
  content: '';
  margin-top: -0.3864em;
  display: table;
}

.sxh0wga-0fxuj {
  padding: 0;
  margin-bottom: var(--sx-space-3);
}

.sx7siaz {
  align-items: center;
  -webkit-appearance: none;
  appearance: none;
  background-color: transparent;
  border: 1px solid var(--sx-colors-tonal500);
  border-radius: var(--sx-radii-round);
  cursor: pointer;
  display: flex;
  justify-content: center;
  padding: 0;
  height: var(--sx-sizes-1);
  width: var(--sx-sizes-1);
  transition: all 50ms ease-out;
}

.sx7siaz:hover {
  outline: none;
}

.sx7siaz:focus {
  outline: none;
}

:checked + .sx7siaz {
  background-color: var(--sx-colors-primary);
  border-color: var(--sx-colors-primary);
}

.sx7siaz[disabled] {
  background-color: var(--sx-colors-tonal200);
  border-color: var(--sx-colors-tonal400);
  cursor: not-allowed;
}

.sxndfzf {
  height: 6px;
  width: 6px;
  border-radius: var(--sx-radii-round);
  background-color: white;
  position: absolute;
}

.sx9ru2a {
  display: flex;
  font-weight: 400;
  max-width: max-content;
}

.sx9ru2a0fgel--align-baseline {
  align-items: baseline;
}

.sx9ru2aged5d--direction-row {
  flex-direction: row;
}

.sxykmo8 {
  display: flex;
  flex-wrap: wrap;
}

.sxykmo88txt2--direction-column {
  flex-direction: column;
}

.sxqqba1 {
  all: unset;
}

<div>
  <form
    aria-label="form"
    class="sx03kze"
  >
    <fieldset
      class="sxqqba1"
    >
      <legend
        class="sxh0wga sxh0wgaryo8u--size-md sxh0wga-0fxuj"
      >
        Example options
      </legend>
      <div
        class="sxykmo8 sxykmo88txt2--direction-column"
        role="radiogroup"
      >
        <div
          class="sx03kze"
        >
          <label
            class="sxh0wga sxh0wgaryo8u--size-md sx9ru2a sx9ru2a0fgel--align-baseline sx9ru2aged5d--direction-row"
          >
            <div
              class="sx03kze sx03kze-2grog"
            >
              <input
                checked=""
                hidden=""
                type="radio"
                value="1"
              />
              <button
                aria-checked="true"
                class="sx7siaz"
                data-radix-roving-focus-group-id="radix-id-0-1"
                data-state="checked"
                role="radio"
                tabindex="0"
                type="button"
                value="1"
              >
                <span
                  class="sxndfzf"
                  data-state="checked"
                />
              </button>
            </div>
            1
          </label>
        </div>
        <div
          class="sx03kze"
        >
          <label
            class="sxh0wga sxh0wgaryo8u--size-md sx9ru2a sx9ru2a0fgel--align-baseline sx9ru2aged5d--direction-row"
          >
            <div
              class="sx03kze sx03kze-2grog"
            >
              <input
                hidden=""
                type="radio"
                value="2"
              />
              <button
                aria-checked="false"
                class="sx7siaz"
                data-radix-roving-focus-group-id="radix-id-0-1"
                data-state="unchecked"
                role="radio"
                tabindex="-1"
                type="button"
                value="2"
              />
            </div>
            2
          </label>
        </div>
      </div>
    </fieldset>
  </form>
</div>
`;
